### PR TITLE
dlg Open NetCDF

### PR DESCRIPTION
**Pull request by stevekogo**

### CODE REVIEW

**Comment by stevekogo** [comment]
@africanmathsinitiative/developers This is ready for review

File: instat/dlgOpenNetCDF.vb [hunk 1]
```diff
@@ -115,13 +116,17 @@ Public Class dlgOpenNetCDF
     End Sub
 
     Private Sub TestOkEnabled()
-        If bCanImport Then
+        If (ucrInputDataName.Text <> "" AndAlso ucrInputLocDataName.Text <> "" AndAlso ucrInputFilePath.Text <> "" AndAlso (Not ucrInputDataName.Text = ucrInputLocDataName.Text)) Then
```
Comment (Lunalo) [inline instat/dlgOpenNetCDF.vb]: We can do gettext here

**Comment by Lunalo** [comment]
We also need to setprefix to avoid overwriting

File: instat/dlgOpenNetCDF.Designer.vb [hunk 2]
```diff
@@ -23,20 +23,103 @@ Partial Class dlgOpenNetCDF
...
+        Me.ucrInputNameLocation.IsReadOnly = False
+        Me.ucrInputNameLocation.Location = New System.Drawing.Point(158, 77)
+        Me.ucrInputNameLocation.Name = "ucrInputNameLocation"
+        Me.ucrInputNameLocation.Size = New System.Drawing.Size(142, 21)
+        Me.ucrInputNameLocation.TabIndex = 15
+        '
+        'lblLocationName
+        '
+        Me.lblLocationName.AutoSize = True
+        Me.lblLocationName.Location = New System.Drawing.Point(14, 82)
```
Comment (Lunalo) [inline instat/dlgOpenNetCDF.Designer.vb]: this should 12 by x

File: instat/dlgOpenNetCDF.Designer.vb [hunk 3]
```diff
@@ -23,20 +23,103 @@ Partial Class dlgOpenNetCDF
...
+        Me.cmdOpenDataSet.Size = New System.Drawing.Size(66, 23)
+        Me.cmdOpenDataSet.TabIndex = 11
+        Me.cmdOpenDataSet.Tag = "select_file"
+        Me.cmdOpenDataSet.Text = "Select File"
+        Me.cmdOpenDataSet.UseVisualStyleBackColor = True
+        '
+        'lblFileOpenPath
+        '
+        Me.lblFileOpenPath.AutoSize = True
+        Me.lblFileOpenPath.Location = New System.Drawing.Point(14, 18)
```
Comment (Lunalo) [inline instat/dlgOpenNetCDF.Designer.vb]: 12 by x

File: instat/dlgOpenNetCDF.Designer.vb [hunk 4]
```diff
@@ -23,20 +23,103 @@ Partial Class dlgOpenNetCDF
...
+        Me.lblFileOpenPath.Name = "lblFileOpenPath"
+        Me.lblFileOpenPath.Size = New System.Drawing.Size(26, 13)
+        Me.lblFileOpenPath.TabIndex = 9
+        Me.lblFileOpenPath.Text = "File:"
+        '
+        'lblSSTName
+        '
+        Me.lblSSTName.AutoSize = True
+        Me.lblSSTName.Location = New System.Drawing.Point(14, 49)
```
Comment (Lunalo) [inline instat/dlgOpenNetCDF.Designer.vb]: 12 by x

**Comment by stevekogo** [comment]
@Lunalo The changes you suggested have been implemented. @africanmathsinitiative/developers can have a look at it.

**Comment by Lunalo** [comment]
@dannyparsons 

I have tested this. It is in a good condition

File: instat/dlgOpenNetCDF.vb [hunk 5]
```diff
@@ -14,36 +14,183 @@
...
+                        ucrInputDataName.SetName(strFileName, bSilent:=True)
+                        ucrInputDataName.Focus()
+                    End If
+                End If
+            End If
+            'TestOkEnabled()
+        End Using
+    End Sub
+
+    Private Sub lblLocationName_Click(sender As Object, e As EventArgs) Handles lblLocationName.Click
```
Comment (lilyclements) [inline instat/dlgOpenNetCDF.vb]: Why are there so many empty subs?

File: instat/dlgOpenNetCDF.Designer.vb [hunk 6]
```diff
@@ -22,21 +22,104 @@ Partial Class dlgOpenNetCDF
...
+        Me.lblSSTName.TabIndex = 3
+        Me.lblSSTName.Tag = "SST_Dataframe_Name:"
+        Me.lblSSTName.Text = "SST Dataframe Name:"
+        '
+        'ucrInputLocDataName
+        '
+        Me.ucrInputLocDataName.AddQuotesIfUnrecognised = True
+        Me.ucrInputLocDataName.IsMultiline = False
+        Me.ucrInputLocDataName.IsReadOnly = False
+        Me.ucrInputLocDataName.Location = New System.Drawing.Point(163, 77)
```
Comment (lilyclements) [inline instat/dlgOpenNetCDF.Designer.vb]: This can be aligned with the other ucrInput so it is at 147, 78

File: instat/dlgOpenNetCDF.vb [hunk 7]
```diff
@@ -14,36 +14,183 @@
 ' You should have received a copy of the GNU General Public License k
 ' along with this program.  If not, see <http://www.gnu.org/licenses/>.
 Imports instat.Translations
+Imports System.IO
+Imports instat
```
Comment (lilyclements) [inline instat/dlgOpenNetCDF.vb]: Imports instat is not needed

**Comment by stevekogo** [comment]
@lilyclements I have a done some housekeeping and the code is neater. However. i cannot have the ucr inputs on the same column position since their labels are not of the same length and the dialog will not look good.

**Comment by lilyclements** [comment]
I suggest we have the two inputs aligned despite the labels being different lengths so that it is like this
image

**Comment by stevekogo** [comment]
@lilyclements  alignment fixed.

**Comment by lilyclements** [comment]
@dannyparsons This is ready for review

**Pull request merged**